python/xorbits/_mars/dataframe/sort/psrs.py

```diff
@@ -665,10 +665,20 @@ def _execute_sort_index_map(cls, ctx, op):
         a, pivots = [ctx[c.key] for c in op.inputs]
         out = op.outputs[0]
 
+        if isinstance(a.index, pd.MultiIndex) and isinstance(pivots, pd.MultiIndex):
+            level = 0 if op.level is None else op.level[0]
```

Comment:
Curr impl may results in long tail execution when the specified level is not evenly distribued. Thus it is needed to find the position using all the dimensions of the multiindex.

Here's an example that could reproduce the problem:
```python
a = np.random.randint(10, size=[2, 100])
a[0] = np.ones(100)
raw_index = pd.MultiIndex.from_arrays(a)

mdf = md.DataFrame(
        {"foo": np.random.randint(100, size=100), "bar": range(100)},
        index=raw_index,
        chunk_size=20
)

mdf.sort_index(
            level=0,
            sort_remaining=True,
)
```

Reply:
done, use `bisect` method to sort multi-level lists rather than reducing dimension and use `searchsorted` method